misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/PartialCoverageModifiedSourceFile.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.runtime.entity;
+
+import org.jacoco.core.analysis.ICounter;
+import org.jacoco.core.analysis.ICoverageNode;
+import org.jacoco.core.analysis.ILine;
+import org.jacoco.core.analysis.ISourceFileCoverage;
+
+import java.util.List;
+
+/**
+ * Represents a source file containing lines modified to consider partially covered coverage info as fully covered.
+ * @since 2.0.0
+ */
+public class PartialCoverageModifiedSourceFile implements ISourceFileCoverage {
+
+    private final ISourceFileCoverage oldSourceFile;
+    private final List<ILine> modifiedLines;
+
+    public PartialCoverageModifiedSourceFile(ISourceFileCoverage oldSourcefile, List<ILine> modifiedLines) {
+        this.oldSourceFile = oldSourcefile;
+        this.modifiedLines = modifiedLines;
+    }
+
+    @Override
+    public int getFirstLine() {
+        return oldSourceFile.getFirstLine();
+    }
+
+    @Override
+    public int getLastLine() {
+        return oldSourceFile.getLastLine();
+    }
+
+    /**
+     * Returns the modified lines instead of lines stored in the original source file.
+     */
+    @Override
+    public ILine getLine(int nr) {
+        if (modifiedLines.size() == 0 || nr < getFirstLine() || nr > getLastLine()) {
```

Comment:
what is the argument name `nr` stands for? please rename it accordingly for readability.

Reply:
It's stands for line number. I'll change it

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/28767/commits/161f96d162389f2a82acd7fe023a0b68e391c46a